src/logic/proposals/encoding.ts

```diff
@@ -170,14 +180,21 @@ export const decodeEvmScript = (script: string, metadata: ProposalMetadata): Dec
     const parameterTypes = metadata.targetSignature
       .substring(metadata.targetSignature.indexOf('(') + 1, metadata.targetSignature.indexOf(')'))
       .split(',');
-    const parameters = utils.defaultAbiCoder.decode(parameterTypes, utils.hexDataSlice(targetCallData, 4));
-    const rawParameters = [...parameters]; // destructuring to enforce Array shape
+    const decodedParameters = utils.defaultAbiCoder.decode(parameterTypes, utils.hexDataSlice(targetCallData, 4));
+    const parameters = await Promise.all(
+      range(parameterTypes.length).map(async (i) => {
+        const param = decodedParameters[i]!;
+        if (parameterTypes[i] !== 'address') return param;
+
+        const ensName = await provider.lookupAddress(param);
+        return ensName || param;
+      })
+    );
 
     return {
       targetAddress: targetContractAddress,
       value,
-      rawParameters,
-      parameters: stringifyBigNumbersRecursively(rawParameters),
+      parameters: stringifyBigNumbersRecursively([...parameters]),
```

Comment:
The "destructuring to enforce Array shape" comment got lost. Is it still necessary to spread here vs just having `stringifyBigNumbersRecursively(parameters)`?

Reply:
I don't think we need that.